[coor/transform] Commute map weighting

maxentile opened the pull request:
Adds an option for tICA to weight eigenvectors following:

[[Noé, Banisch, Clementi, 2016] Commute maps: separating slowly-mixing molecular configurations for kinetic modeling](http://pubs.acs.org/doi/abs/10.1021/acs.jctc.6b00762)

maxentile committed:
Merge remote-tracking branch 'markovmodel/devel' into devel

maxentile committed:
Add commute_map option

- Weights each eigenvector_i by sqrt(timescale_i / 2), as in
http://pubs.acs.org/doi/abs/10.1021/acs.jctc.6b00762
- If commute_map=True and kinetic_map=True, raise a ValueError

maxentile committed:
Regularize timescales

Dampens timescales smaller than the lag (c.f. section 2.5 of ref. [5])

franknoe commented:
Thanks. kinetic_map and commute_map should be mutually exclusive, i.e. when both set to True it should rather raise, and selection should be done with elif.

franknoe commented:
Also, I guess it would help to have the option in the coordinates API function too.

maxentile committed:
Raise ValueError in `transform_array`

Previously, this was only checked in the constructor. This now catches
if the user has set the `kinetic_map` or `commute_map` attributes
incorrectly after construction.

maxentile committed:
Add journal reference for kinetic maps paper

maxentile committed:
Add commute_map option in coordinates API function

franknoe commented:
Thanks Josh, this looks good!
